### PR TITLE
Update packages.

### PR DESCRIPTION
**Pull request by NN---**
Update packages to newer versions which has .NET Standard build, so it can be consumed by .NET Core.
Fix using obsolete methods.

**I grant this project or any other developer the permissions to use unconditionally the provided code without any limitation and to use it for any purpose.**

### CODE REVIEW

**Comment by gerhardol** [comment]
For next release, I assume.
Will have to be rebased on master, for instance #8123 updates RestSharp including some code changes.
The submodules need to be updated too. (Versions must match at least, I assume that there are dependencies too.)

**Comment by NN---** [comment]
Yeah, submodules needs to be updated as well.

**Comment by NN---** [comment]
Btw is there a reason we use submodules instead of putting in this repo ?
It makes PRs much easier to maintain and to propose.

**Comment by gerhardol** [comment]
> Btw is there a reason we use submodules instead of putting in this repo ?

The submodules are cloned from other origins, by keeping the structure it is easier to take updates and push changes upstream. Not much used for the GE submodules though.
Doc is of course special, EasyHook is a submodule so we can drop that code when no longer needed (after some .NET and Winforms updates in the future).
The editor is a separate .exe too. NBug was actually moved into GE when most functionality was removed.

Subtrees could be used instead, no major change.

Submodules is the standard way in Git to share codem this will not likely change.

**Comment by RussKie** [comment]
Thank you. I'll get to it once we push v3.4 out of the door.

**Comment by RussKie** [comment]
@NN--- can you please rebase and resolve the MC

**Comment by NN---** [comment]
Ok

**Comment by NN---** [comment]
This change required Externals/Git.hub fix

**Comment by RussKie** [comment]
> This change required Externals/Git.hub fix

Please send a PR to it

**Comment by NN---** [comment]
Added PR

**Comment by pmiossec** [comment]
@NN--- I fixed the unit tests not passing. Rebase and squash when you review my commit.

**Comment by NN---** [comment]
@pmiossec GitHub has squash merge which does essentially the same, no ?

**Comment by pmiossec** [comment]
@NN--- Yes but that let the responsibility to the maintainer to squash and choose the best commit message where the contributor is sometimes the best placed to do it.

That's why if the PR could be in a finished state, where the maintainer just has to merge or rebase and merge, it's easier for everyone...

File: GitExtensions.ruleset [hunk 1]
```diff
@@ -48,9 +48,11 @@
     <Rule Id="SA1403" Action="None" />
     <Rule Id="SA1404" Action="None" />
     <Rule Id="SA1413" Action="None" />
+    <Rule Id="SA1503" Action="None" />
     <Rule Id="SA1504" Action="None" />
     <Rule Id="SA1512" Action="None" />
     <Rule Id="SA1516" Action="None" />
+    <Rule Id="SA1519" Action="None" />
```
Comment (RussKie) [inline GitExtensions.ruleset:55]: Why are disabling these rules?

File: GitUI/GitUIExtensions.cs [hunk 2]
```diff
@@ -110,7 +110,9 @@ void OpenWithDiffTool()
                 if (file.IsSubmodule && file.GetSubmoduleStatusAsync() != null)
                 {
                     // Patch already evaluated
+#pragma warning disable VSTHRD103
                     var status = ThreadHelper.JoinableTaskFactory.Run(file.GetSubmoduleStatusAsync);
```
Comment (RussKie) [inline GitUI/GitUIExtensions.cs:114]: Can you please provide more details on this supression?

Comment (NN---) [reply GitUI/GitUIExtensions.cs:114]: Same way as suppression here:
https://github.com/gitextensions/gitextensions/pull/7747/files#diff-fa0461ff6916a00b8ca6742fc94d5bd5R182

File: GitExtensions.ruleset (hunk 1 above)
Comment (NN---) [reply GitExtensions.ruleset:55]: Seems like not needed.
Removing

File: GitExtensions.ruleset [hunk 3]
```diff
@@ -48,6 +48,7 @@
     <Rule Id="SA1403" Action="None" />
     <Rule Id="SA1404" Action="None" />
     <Rule Id="SA1413" Action="None" />
+    <Rule Id="SA1503" Action="None" />
```
Comment (RussKie) [inline GitExtensions.ruleset:51]: What about this one?

File: GitExtensions.ruleset (hunk 1 above)
Comment (NN---) [reply GitExtensions.ruleset]: Fixed by adding braces as expected by the rule: https://github.com/gitextensions/gitextensions/pull/8156/files#diff-0c448da7cf150a550546e895757d16c2R182

File: GitUI/GitUIExtensions.cs (hunk 2 above)
Comment (RussKie) [reply GitUI/GitUIExtensions.cs:114]: Please add comments explaining the supressions, like in the other commit.

Comment (NN---) [reply GitUI/GitUIExtensions.cs:114]: Done

File: GitExtensions.ruleset (hunk 3 above)
Comment (NN---) [reply GitExtensions.ruleset]: Sorry, forgot to remove.

**Comment by RussKie** [comment]
Can you please provide sign-off in contributors.txt? Without it things get complicated as far as licenses concerned

File: Solution.Versions.props [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
-    <BenDemystifierVersion>0.1.4</BenDemystifierVersion>
-    <JetBrainsAnnotationsVersion>2018.2.1</JetBrainsAnnotationsVersion>
-    <LibGit2SharpVersion>0.25.0</LibGit2SharpVersion>
-    <MicrosoftVisualStudioCompositionVersion>15.6.36</MicrosoftVisualStudioCompositionVersion>
-    <MicrosoftVisualStudioThreadingVersion>16.5.132</MicrosoftVisualStudioThreadingVersion>
-    <NewtonsoftJsonVersion>10.0.3</NewtonsoftJsonVersion>
+    <AdysTechCredentialManagerVersion>2.1.1</AdysTechCredentialManagerVersion>
+    <AppInsightsWindowsDesktopVersion>2.13.1</AppInsightsWindowsDesktopVersion>
+    <BenDemystifierVersion>0.1.6</BenDemystifierVersion>
+    <JetBrainsAnnotationsVersion>2020.1.0 </JetBrainsAnnotationsVersion>
```
Comment (RussKie) [inline Solution.Versions.props:7]: Just noticed stray spaces in few places throughout the file
```suggestion
    <JetBrainsAnnotationsVersion>2020.1.0</JetBrainsAnnotationsVersion>
```

Comment (NN---) [reply Solution.Versions.props:7]: 👍

**Comment by mstv** [comment]
@NN--- has granted the free use of these changes. So one who signed the CLA, could adopt the changes and create an identical PR.
We won't do this often but it seems to be worth in this case. Shall we proceed this way?

**Comment by RussKie** [comment]
I've taken @NN--- changes in the .NET Core branch.

**Comment by gerhardol** [comment]
> I've taken @NN--- changes in the .NET Core branch.

@RussKie 
Will .NET 5 be merged soon, should we make a separate PR for this in master or keep this open?

**Comment by RussKie** [comment]
This was incorporated into #8522. Thank you.

**Pull request closed without merging**